samtranslator/model/stepfunctions/events.py

```diff
@@ -50,7 +51,13 @@ def _generate_logical_id(self, prefix, suffix, resource_type):  # type: ignore[n
         generator = logical_id_generator.LogicalIdGenerator(prefix + resource_type, suffix)
         return generator.gen()
 
-    def _construct_role(self, resource, permissions_boundary=None, prefix=None, suffix=""):  # type: ignore[no-untyped-def]
+    def _construct_role(
+        self,
+        resource: StepFunctionsStateMachine,
+        permissions_boundary: Optional[str],
+        prefix: Optional[str],
+        suffix: str = "",
```

Comment:
why `str = ""` here rather than `Optional[str]`

Reply:
https://github.com/aws/serverless-application-model/pull/3363/files#diff-e4eab90ce32328ebbbb90bad6d4e5ea21e06f241708a08eb9354c88146301f9dR49 it checks whether the suffix is alphanumerical but doesnt do that for the prefix. So I tried to avoid changing the logic there and just left it to default to the empty string.

Reply:
Talked offline and decided that changing _generate_logical_id() was not worth it